inspirehep/modules/multieditor/api.py

```diff
@@ -0,0 +1,160 @@
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Module for backend of multi record editor used in http://inspirehep.net."""
+
+from __future__ import absolute_import, print_function, division
+
+from flask import Blueprint, request, jsonify, session
+import copy
+
+from inspire_schemas.api import load_schema
+from inspirehep.modules.multieditor import tasks
+from inspirehep.modules.migrator.tasks import chunker
+from .actions import ActionFactory, compare_records
+from . import queries
+from .permissions import multieditor_use_permission
+
+
+match_type_map = {
+    'contains': 'contains',
+    'is equal to': 'exact',
+    'does not exist': 'missing',
+    'matches regular expression': 'regex'
+}
+
+blueprint = Blueprint(
+    'inspirehep_multieditor',
+    __name__,
+    template_folder='templates',
+    url_prefix='/multieditor',
+)
+
+
+@blueprint.route("/update", methods=['POST'])
+@multieditor_use_permission.require(http_exception=403)
+def update():
+    """Apply the user actions to the database records."""
+    user_actions = request.json['userActions']
+    all_selected = request.json['allSelected']
+    multieditor_session = session.get('multieditor_session', [])
+    if multieditor_session:
+        ids = multieditor_session['ids']
+        if all_selected:
+            ids = set(ids) - set(request.json['ids'])
+        else:
+            ids = set(ids) & set(request.json['ids'])
+    else:
+        return jsonify({'message': 'Please use the search before you apply actions'}), 400
+    actions = []
+    conditions = sanitize_user_conditions(user_actions.get('conditions', []))
+
+    for action in user_actions.get('actions', []):
+        actions.append(
+            ActionFactory.factory(action.get('actionName', ''),
+                                  **sanitize_user_action(action, conditions))
+        )
+    for i, chunk in enumerate(chunker(ids, 20)):
+        tasks.process_records.delay(records_ids=chunk, actions=actions, schema=multieditor_session['schema'])
+    return jsonify({'message': 'Records are being updated'})
+
+
+@blueprint.route("/preview", methods=['POST'])
+@multieditor_use_permission.require(http_exception=403)
+def preview():
+    """Preview the user actions in the first (page size) records."""
+    user_actions = request.json['userActions']
+    query_string = request.json['queryString']
+    page_size = int(request.json['pageSize'])
+    page_num = request.json['pageNum']
+    multieditor_session = session.get('multieditor_session', [])
+
+    if not multieditor_session:
+        return jsonify({'message': 'Please use the search before you apply actions'}), 400
+
+    records = queries.get_records_from_query(query_string, page_size, page_num, multieditor_session['schema_name'])['json_records']
```

Comment:
Above https://github.com/inspirehep/inspire-next/pull/2669/files#diff-6a582a185e685dc4bd77e3504aba3be7R79 you use `multieditor_session['schema']`, what is each used for?